include/sys/zfs_context.h

```diff
@@ -766,4 +766,7 @@ extern void spl_fstrans_unmark(fstrans_cookie_t);
 extern int spl_fstrans_check(void);
 
 #endif /* _KERNEL */
+
+#define	ZFS_ARRAY_SIZE(a)	(sizeof (a) / sizeof (*a))
```

Comment:
Are there different implementations for userland (libzpool) vs kernel?  If not, this probably doesn't belong in zfs_context.h.